ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py

```diff
@@ -530,46 +467,46 @@ def test_torch_matrix_exp(
         min_num_dims=2,
         min_axes_size=2,
         max_axes_size=2,
+        min_value=-1e04,
+        max_value=1e04,
         valid_axis=True,
         force_tuple_axis=True,
     ),
     ord=st.sampled_from(["fro", "nuc", np.inf, -np.inf, 1, -1, 2, -2]),
     keepdim=st.booleans(),
-    dtype=helpers.get_dtypes("float", none=True, full=False),
 )
 def test_torch_matrix_norm(
-    *,
-    dtype_values_axis,
-    ord,
-    keepdim,
-    dtype,
-    frontend,
-    test_flags,
-    fn_tree,
-    on_device,
+        *,
+        dtype_values_axis,
+        ord,
+        keepdim,
+        frontend,
+        test_flags,
+        fn_tree,
+        on_device,
 ):
     input_dtype, x, axis = dtype_values_axis
+
     helpers.test_frontend_function(
         input_dtypes=input_dtype,
         frontend=frontend,
         test_flags=test_flags,
         fn_tree=fn_tree,
         on_device=on_device,
-        # rtol=1e-04,
-        # atol=1e-04,
+        rtol=1e-03,
+        atol=1e-03,
         input=x[0],
         ord=ord,
         dim=axis,
         keepdim=keepdim,
-        dtype=dtype[0],
```

Comment:
Ah yes correct!

This was a complicated case. Because the "dtype" cannot be a dtype with less bits the the input dtype, and it can only be a float dtype for float inputs. So the logic behind generating values for this argument when testing for all valid dtypes is not trivial. I'll revert the changes and update this test in another PR, where I will have to implement that logic in a st.composite function

Reply:
I've seen this problem before. Currently we have `helpers.get_castable_dtype` but it doesn't cover such cases.
It would be great if you could make the strategy general enough to handle [this](https://github.com/unifyai/ivy/blob/183173a29a1362ba5a8dc13f4828761c9026c2b2/ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py#L885) case as well!